src/lib/operators/table_scan.cpp

```diff
@@ -0,0 +1,247 @@
+#include <memory>
+#include <numeric>
+#include <optional>
+#include <storage/dictionary_segment.hpp>
+#include <storage/reference_segment.hpp>
+#include <storage/value_segment.hpp>
+#include <string>
+#include <vector>
+
+#include "abstract_operator.hpp"
+#include "all_type_variant.hpp"
+#include "operators/print.hpp"
+#include "resolve_type.hpp"
+#include "storage/table.hpp"
+#include "table_scan.hpp"
+#include "types.hpp"
+#include "utils/assert.hpp"
+
+namespace /* anonymous */ {
```

Comment:
It's to prevent the `get_comparator(ScanType)` functions from being part of the TableScan class and to limit their visibility to the table_scan.cpp file